trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/credentials/Signer.java

```diff
@@ -111,10 +96,15 @@ static String sign(
 
         queryParameters.forEach(requestBuilder::putRawQueryParameter);
 
-        Aws4SignerParams.Builder<?> signerParamsBuilder = Aws4SignerParams.builder()
+        boolean enablePayloadSigning = Optional.of(requestHeaders.getFirst("x-amz-content-sha256"))
```

Comment:
What about

```java
        boolean enablePayloadSigning = !("UNSIGNED-PAYLOAD".equals(requestHeaders.getFirst("x-amz-content-sha256")));

```

Alternatively, if we want to keep this optional one, this should be `Optional.ofNullable` right?

Reply:
Fixed to `Optional.ofNullable`